File: mobile/ios/Actnet/Sources/Utils/AvalancheColors.swift
import SwiftUI

extension Color {
    // MARK: - Plum (primary brand scale)
    static let plum50  = Color(red: 0.945, green: 0.922, blue: 0.929) // #F1EBED
    static let plum100 = Color(red: 0.863, green: 0.784, blue: 0.808) // #DCC8CE
    static let plum200 = Color(red: 0.784, green: 0.706, blue: 0.741) // #C8B4BD
    static let plum300 = Color(red: 0.694, green: 0.584, blue: 0.624) // #B1959F
    static let plum400 = Color(red: 0.612, green: 0.463, blue: 0.510) // #9C7682
    static let plum500 = Color(red: 0.420, green: 0.243, blue: 0.314) // #6B3E50
    static let plum600 = Color(red: 0.361, green: 0.204, blue: 0.271) // #5C3445
    static let plum700 = Color(red: 0.302, green: 0.161, blue: 0.224) // #4D2939
    static let plum800 = Color(red: 0.235, green: 0.122, blue: 0.176) // #3C1F2D
    static let plum900 = Color(red: 0.165, green: 0.086, blue: 0.125) // #2A1620

    // MARK: - Sand (warm neutral)
    static let sand50  = Color(red: 1.000, green: 0.973, blue: 0.945) // #FFF8F1
    static let sand100 = Color(red: 1.000, green: 0.945, blue: 0.914) // #FFF1E9
    static let sand200 = Color(red: 0.957, green: 0.906, blue: 0.851) // #F4E7D9
    static let sand300 = Color(red: 0.898, green: 0.831, blue: 0.753) // #E5D4C0
    static let sand400 = Color(red: 0.780, green: 0.714, blue: 0.627) // #C7B6A0
    static let sand500 = Color(red: 0.604, green: 0.541, blue: 0.471) // #9A8A78
    static let sand600 = Color(red: 0.431, green: 0.384, blue: 0.345) // #6E6258
    static let sand900 = Color(red: 0.122, green: 0.094, blue: 0.082) // #1F1815

    // MARK: - Accents (reserved, status only)
    static let rose400 = Color(red: 0.820, green: 0.459, blue: 0.475) // #D17579
    static let rose500 = Color(red: 0.780, green: 0.408, blue: 0.439) // #C76870
    static let rose700 = Color(red: 0.600, green: 0.243, blue: 0.278) // #993E47
    static let moss    = Color(red: 0.420, green: 0.498, blue: 0.302) // #6B7F4D
    static let amber   = Color(red: 0.831, green: 0.627, blue: 0.290) // #D4A04A

    // MARK: - Semantic aliases
    static let avBrand          = Color.plum500
    static let avPaper          = Color.sand100
    static let avInk            = Color.sand900
    static let avMuted          = Color.sand600
    static let avOutgoingBubble = Color.plum500
    static let avIncomingBubble = Color.sand200
    static let avDarkSurface    = Color.plum900
    static let avError          = Color.rose700
    static let avNotification   = Color.rose500
    static let avSuccess        = Color.moss
    static let avWarning        = Color.amber
}

File: mobile/android/app/src/main/kotlin/Utils/AvalancheColors.kt
package net.theavalanche.app

import android.content.Context
import android.content.res.Configuration
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color

// Brand palette. Mirrors mobile/ios/Actnet/Sources/Utils/AvalancheColors.swift —
// keep the two in sync (cross-platform parity rule).
//
// The raw palette (Plum*/Sand*/accents) below is fixed reference swatches. The
// *semantic* colors are dark-mode-adaptive and live in AvalancheSemanticColors /
// LocalAvalancheColors at the bottom of this file — Composable views read those
// (e.g. LocalAvalancheColors.current.paper), mirroring iOS's adaptive Color
// aliases. The `Brand`/`Paper`/... aliases on the object are kept as the *light*
// values for the MaterialTheme scheme (Theme.kt), QR rendering, and previews.
object AvalancheColors {
    // MARK: - Plum (primary brand scale)
    val Plum50 = Color(0xFFF1EBED)
    val Plum100 = Color(0xFFDCC8CE)
    val Plum200 = Color(0xFFC8B4BD)
    val Plum300 = Color(0xFFB1959F)
    val Plum400 = Color(0xFF9C7682)
    val Plum500 = Color(0xFF6B3E50)
    val Plum600 = Color(0xFF5C3445)
    val Plum700 = Color(0xFF4D2939)
    val Plum800 = Color(0xFF3C1F2D)
    val Plum900 = Color(0xFF2A1620)

    // MARK: - Sand (warm neutral)
    val Sand50 = Color(0xFFFFF8F1)
    val Sand100 = Color(0xFFFFF1E9)
    val Sand200 = Color(0xFFF4E7D9)
    val Sand300 = Color(0xFFE5D4C0)
    val Sand400 = Color(0xFFC7B6A0)
    val Sand500 = Color(0xFF9A8A78)
    val Sand600 = Color(0xFF6E6258)
    val Sand900 = Color(0xFF1F1815)

    // MARK: - Accents (reserved, status only)
    val Rose400 = Color(0xFFD17579)
    val Rose500 = Color(0xFFC76870)
    val Rose700 = Color(0xFF993E47)
    val Moss = Color(0xFF6B7F4D)
    val Amber = Color(0xFFD4A04A)

    // MARK: - Semantic aliases
    val Brand = Plum500
    val Paper = Sand100
    val Ink = Sand900
    val Muted = Sand600
    val OutgoingBubble = Plum500
    val IncomingBubble = Sand200
    val DarkSurface = Plum900
    val Error = Rose700
    val Notification = Rose500
    val Success = Moss
    val Warning = Amber
}

/// Dark-mode-adaptive semantic colors. Mirrors the adaptive `av*` aliases in the
/// iOS AvalancheColors.swift. Warm "plum-black" dark theme: deep aubergine
/// surfaces, warm off-white text.
data class AvalancheSemanticColors(
    val brand: Color,
    val paper: Color,
    val ink: Color,
    val muted: Color,
    val incomingBubble: Color,
    val outgoingBubble: Color,
    val card: Color,
    val divider: Color,
    val error: Color,
    val notification: Color,
    val success: Color,
    val warning: Color,
)

val LightAvalancheColors = AvalancheSemanticColors(
    brand = AvalancheColors.Plum500,
    paper = AvalancheColors.Sand100,
    ink = AvalancheColors.Sand900,
    muted = AvalancheColors.Sand600,
    incomingBubble = AvalancheColors.Sand200,
    outgoingBubble = AvalancheColors.Plum500,
    card = AvalancheColors.Sand50,
    divider = AvalancheColors.Sand300,
    error = AvalancheColors.Rose700,
    notification = AvalancheColors.Rose500,
    success = AvalancheColors.Moss,
    warning = AvalancheColors.Amber,
)

val DarkAvalancheColors = AvalancheSemanticColors(
    brand = AvalancheColors.Plum200,
    paper = AvalancheColors.Plum900,
    ink = AvalancheColors.Sand50,
    muted = AvalancheColors.Sand300,
    incomingBubble = AvalancheColors.Plum800,
    outgoingBubble = AvalancheColors.Plum500, // always on plum; stands out in both modes
    card = AvalancheColors.Plum800,
    divider = AvalancheColors.Plum700,
    error = AvalancheColors.Rose400,
    notification = AvalancheColors.Rose500,
    success = AvalancheColors.Moss,
    warning = AvalancheColors.Amber,
)

/// Provided by AvalancheTheme based on the system light/dark setting. Composables
/// read `LocalAvalancheColors.current.<role>`.
val LocalAvalancheColors = staticCompositionLocalOf { LightAvalancheColors }

/// Resolves the semantic palette outside a Composable scope (e.g. inside a custom
/// android.view.View such as RecipientTokenField) from the context's night-mode
/// configuration. The hosting Activity is recreated on a uiMode change, so this is
/// re-read with the correct values after a light/dark toggle.
fun avalancheSemanticColors(context: Context): AvalancheSemanticColors {
    val night = (context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) ==
        Configuration.UI_MODE_NIGHT_YES
    return if (night) DarkAvalancheColors else LightAvalancheColors
}
